src/hooks/useEtherspotSwaps.ts

```diff
@@ -116,17 +123,57 @@ const useEtherspotSwaps = (chainId?: number): IEtherspotSwapsHook => {
       return { type: 'same-chain', offers };
     } catch (e) {
       console.warn(
-        `Sorry, an error occurred whilst trying to fetch same-chain offers.`
-        + ` Please try again. Error:`,
-        e,
+        `Sorry, an error occurred whilst trying to fetch same-chain offers.` + ` Please try again. Error:`,
+        e
+      );
+    }
+  };
+
+  const getQuotes = async (
+    fromAccountAddress: string,
+    toAddress: string,
+    toChainId: number,
+    fromToken: string,
+    fromAmount: BigNumber,
+    slippage: number,
+    provider?: BridgingProvider
+  ): Promise<Quote[] | undefined> => {
+    const sdkForChainId = await getSdk(swapsChainId);
+
+    const fromAccount = fromAccountAddress ?? (await sdkForChainId.getCounterFactualAddress());
+    if (!fromAccount) {
+      console.warn(`No account address provided!`);
+      return;
+    }
+
+    const dataService = getDataService();
+
+    try {
+      const quotes = await dataService.getQuotes({
+        fromAddress: fromAccount,
+        toAddress,
+        fromChainId: +swapsChainId,
+        toChainId,
+        fromToken,
+        fromAmount,
+        slippage,
+        provider,
+      });
+
+      return quotes;
+    } catch (e) {
+      console.warn(
+        `Sorry, an error occurred whilst trying to fetch cross-chain quotes.` + ` Please try again. Error:`,
+        e
       );
     }
```

Comment:
Unless i'm missing something we should return an empty array here in the case of an error, or throw an error

Reply:
should we not warn that there is an error though? if empty array, the user will think that there is no quotes for what they are asking for, rather being an error?

Reply:
Sorry just to be clear - i meant both return an empty array **in addition to** the warning